WooCommerce/Classes/Extensions/String+Helpers.swift

```diff
@@ -68,4 +68,150 @@ extension String {
     var isNotEmpty: Bool {
         return !isEmpty
     }
+
+    /// Compares two strings as versions.
+    ///
+    /// Returns `orderedAscending` if the argument represents a newer version
+    /// Returns `orderedSame` if the argument represents the same version
+    /// Returns `orderedDescending` if the argument represents an older version
+    ///
+    func compareAsVersion(to: String) -> ComparisonResult {
+
+        /// Replace _ - and + with a .
+        ///
+        func replaceUnderscoreDashAndPlusWithDot(inString: String) -> String {
+            var outString = inString
+            let replaceTargets = ["_", "-", "+"]
+            for replaceTarget in replaceTargets {
+                outString = outString.replacingOccurrences(of: replaceTarget, with: ".")
+            }
+            return outString
+        }
```

Comment:
nit:

Perhaps these can be simplified by using regular expression? I haven't tested it yet, something like:

```suggestion
        func replaceUnderscoreDashAndPlusWithDot(_ string: String) -> String {
            string.replacingOccurrences(of: "[_-+]", with: ".", options: .regularExpression)
        }
```

Reply:
Works with a little tweak to the regex. Fixed in https://github.com/woocommerce/woocommerce-ios/pull/5382/commits/1e7b070116a4f919006e42166e5836d305331c55